### ISSUE TITLE
Mixer v2 issues

### ISSUE THREAD

**Issue by PiotrSikora** [envoyproxy/envoy-wasm#120]
@bianpengyuan could you please write down the issues you've run into that resulted in Mixer v2 using null plugin instead of proper WASM runtime? Thanks!

**Comment by bianpengyuan** [comment]
Ack, I need to do quite a context switching to reproduce the issue.  Will try to get to this in the next two days.

**Comment by PiotrSikora** [comment]
Ping.

**Comment by bianpengyuan** [comment]
Blocked on https://github.com/envoyproxy/envoy-wasm/issues/136

**Comment by PiotrSikora** [comment]
Now that #136 is fixed, could you try to reproduce your issues? Thanks!

**Comment by bianpengyuan** [comment]
Hit another issue when trying to reproduce:
```
[2019-08-23 15:35:08.700][107768][trace][wasm] [source/extensions/common/wasm/v8/v8.cc:342] [wasm] link(), export host global: env.DYNAMICTOP_PTR (i32)
[2019-08-23 15:35:08.700][107768][trace][wasm] [source/extensions/common/wasm/v8/v8.cc:342] [wasm] link(), export host global: global.NaN (f64)
[2019-08-23 15:35:08.700][107768][trace][wasm] [source/extensions/common/wasm/v8/v8.cc:342] [wasm] link(), export host global: global.Infinity (f64)
[2019-08-23 15:35:08.700][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:81] Caught Segmentation fault, suspect faulting address 0x0
[2019-08-23 15:35:08.700][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:69] Backtrace (use tools/stack_decode.py to get line numbers):
[2019-08-23 15:35:08.700][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #0: __restore_rt [0x7f8154f4d3a0]
[2019-08-23 15:35:08.715][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #1: v8::internal::LookupIterator::PrepareTransitionToDataProperty() [0x87798b]
[2019-08-23 15:35:08.724][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #2: v8::internal::Object::AddDataProperty() [0x899098]
[2019-08-23 15:35:08.731][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #3: v8::internal::Object::SetProperty() [0x8982de]
[2019-08-23 15:35:08.738][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #4: wasm::Instance::make() [0x69b37e]
[2019-08-23 15:35:08.743][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #5: Envoy::Extensions::Common::Wasm::V8::V8::link() [0x67b3bb]
[2019-08-23 15:35:08.749][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #6: Envoy::Extensions::Common::Wasm::Wasm::initialize() [0x66b1e3]
[2019-08-23 15:35:08.754][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #7: Envoy::Extensions::Common::Wasm::createWasmInternal() [0x66d741]
[2019-08-23 15:35:08.758][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #8: Envoy::Extensions::Common::Wasm::createWasm() [0x66d415]
[2019-08-23 15:35:08.763][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #9: Envoy::Extensions::HttpFilters::Wasm::FilterConfig::FilterConfig() [0x655ec3]
[2019-08-23 15:35:08.768][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #10: Envoy::Extensions::HttpFilters::Wasm::WasmFilterConfig::createFilterFactoryFromProtoTyped() [0x653458]
[2019-08-23 15:35:08.774][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #11: Envoy::Extensions::HttpFilters::Common::FactoryBase<>::createFilterFactoryFromProto() [0x65374d]
[2019-08-23 15:35:08.779][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #12: Envoy::Extensions::NetworkFilters::HttpConnectionManager::HttpConnectionManagerConfig::processFilter() [0xf6ccf6]
[2019-08-23 15:35:08.784][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #13: Envoy::Extensions::NetworkFilters::HttpConnectionManager::HttpConnectionManagerConfig::HttpConnectionManagerConfig() [0xf6bb93]
[2019-08-23 15:35:08.789][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #14: Envoy::Extensions::NetworkFilters::HttpConnectionManager::HttpConnectionManagerFilterConfigFactory::createFilterFactoryFromProtoTyped() [0xf697d5]
[2019-08-23 15:35:08.794][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #15: Envoy::Extensions::NetworkFilters::Common::FactoryBase<>::createFilterFactoryFromProto() [0xf6dda5]
[2019-08-23 15:35:08.799][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #16: Envoy::Server::ProdListenerComponentFactory::createNetworkFilterFactoryList_() [0x12564c0]
[2019-08-23 15:35:08.804][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #17: Envoy::Server::ProdListenerComponentFactory::createNetworkFilterFactoryList() [0x1260aa4]
[2019-08-23 15:35:08.809][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #18: Envoy::Server::ListenerFilterChainFactoryBuilder::buildFilterChain() [0x126080c]
[2019-08-23 15:35:08.814][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #19: Envoy::Server::FilterChainManagerImpl::addFilterChain() [0x1265d66]
[2019-08-23 15:35:08.819][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #20: Envoy::Server::ListenerImpl::ListenerImpl() [0x125a64b]
[2019-08-23 15:35:08.824][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #21: Envoy::Server::ListenerManagerImpl::addOrUpdateListener() [0x125ddc1]
[2019-08-23 15:35:08.828][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #22: Envoy::Server::Configuration::MainImpl::initialize() [0x1285fad]
[2019-08-23 15:35:08.834][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #23: Envoy::Server::InstanceImpl::initialize() [0x12467bf]
[2019-08-23 15:35:08.839][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #24: Envoy::Server::InstanceImpl::InstanceImpl() [0x1242404]
[2019-08-23 15:35:08.844][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #25: std::make_unique<>() [0x449423]
[2019-08-23 15:35:08.849][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #26: Envoy::MainCommonBase::MainCommonBase() [0x44911d]
[2019-08-23 15:35:08.854][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #27: Envoy::MainCommon::MainCommon() [0x4499f5]
[2019-08-23 15:35:08.858][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #28: main [0x448713]
[2019-08-23 15:35:08.859][107768][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #29: __libc_start_main [0x7f8154b9f52b]
```

Does this ring a bell to you?

**Comment by PiotrSikora** [comment]
This looks like an old issue that I thought was fixed already (and I removed a patch for it when updating from v7.5 to v7.6).

Could you rebuild at https://github.com/envoyproxy/envoy-wasm/commits/69395dc925e2e4f563a834135b9b755126b116be and see if that happens there as well?

**Comment by bianpengyuan** [comment]
Yeah it still happens to me after recompile from head b7827d9cbc11f82

**Comment by bianpengyuan** [comment]
In case you want to try out, there is the wasm file: https://github.com/bianpengyuan/wasm-sd/blob/compiled-in-and-wasm/wasm-sd.wasm

**Comment by PiotrSikora** [comment]
> Yeah it still happens to me after recompile from head [b7827d9](https://github.com/envoyproxy/envoy-wasm/commit/b7827d9cbc11f8296e1b038dc94c4fa307b9c143)

Erm, I've asked you to check an older commit (https://github.com/envoyproxy/envoy-wasm/commit/69395dc925e2e4f563a834135b9b755126b116be) to see if that's a regression from upgrading from v7.5 to v7.6... Anyway, I tested it myself, and it looks that v7.6 update is indeed the issue.

**Comment by PiotrSikora** [comment]
Hmm, I thought that maybe there was an issue with `global.Infinity` import, but it seems that it works fine (see: https://github.com/envoyproxy/envoy-wasm/pull/150).

**Comment by bianpengyuan** [comment]
I see, I rebuilt 69395dc and I can no longer reproduce the issue. The filter works well under high load. Feel free to close the issue, thanks!

**Comment by PiotrSikora** [comment]
I found the issue, it's in your code: https://github.com/bianpengyuan/wasm-sd/pull/1, not in V8.

Basically, you used an invalid option (at least I hope so), and instead of defining maximum memory size, you defined minimum memory size... to 512MB, and V8 didn't like.

Please rebuild your extension and test again at current `master`. Thanks!

**Comment by bianpengyuan** [comment]
Thanks! I tested and it works with head.

**Comment by PiotrSikora** [comment]
Did you test it under high load as well? If so, can I close this issue?

**Comment by bianpengyuan** [comment]
Yes, I tested with high load. No issue was found this time.

**Comment by PiotrSikora** [comment]
Awesome, thanks!
